fireshape/control.py

```diff
@@ -138,21 +146,51 @@ def __init__(self, mesh_r):
         self.T.assign(self.id)
         self.mesh_m = fd.Mesh(self.T)
         self.V_m = fd.FunctionSpace(self.mesh_m, element)
+        self.is_DG = False
+
+        """
+        ControlSpace for discontinuous coordinate fields
+        (e.g.  periodic domains)
+
+        In Firedrake, periodic meshes are implemented using a discontinuous
+        field. This implies that self.V_r contains discontinuous functions.
+        To ensure domain updates do not create holes in the domain,
+        use a continuous subspace self.V_c of self.V_r as control space.
+        """
+        if element.family() == 'Discontinuous Lagrange':
+            self.is_DG = True
+            self.V_c = fd.VectorFunctionSpace(self.mesh_r,
+                                              "CG", element._degree)
+            self.Ip = fd.Interpolator(fd.TestFunction(self.V_c),
+                                      self.V_r).callable().handle
```

Comment:
if we add` else: self.V_c = self.V_r`, we can spare a few lines in `get_zero_vec` and `get_space_for_inner`

I'll let @florianwechsung  decide

Reply:
I think its fine as is, no strong opinion here.